Firebase/Auth/Source/AuthProviders/OAuth/FIROAuthCredential.m

```diff
@@ -40,30 +41,50 @@ - (nullable instancetype)initWithProvider:(NSString *)provider {
 - (instancetype)initWithProviderID:(NSString *)providerID
                            IDToken:(nullable NSString *)IDToken
                        accessToken:(nullable NSString *)accessToken
-                      pendingToken:(nullable NSString *)pendingToken {
+                      pendingToken:(nullable NSString *)pendingToken
+                            secret:(nullable NSString *)secret {
   self = [super initWithProvider:providerID];
   if (self) {
     _IDToken = IDToken;
     _accessToken = accessToken;
     _pendingToken = pendingToken;
+    _secret = secret;
   }
   return self;
 }
 
 - (instancetype)initWithProviderID:(NSString *)providerID
                          sessionID:(NSString *)sessionID
             OAuthResponseURLString:(NSString *)OAuthResponseURLString {
-  self = [self initWithProviderID:providerID IDToken:nil accessToken:nil pendingToken:nil];
+  self =
+      [self initWithProviderID:providerID IDToken:nil accessToken:nil pendingToken:nil secret:nil];
   if (self) {
     _OAuthResponseURLString = OAuthResponseURLString;
     _sessionID = sessionID;
   }
   return self;
 }
 
+
+- (nullable instancetype)initWithVerifyAssertionResponse:(FIRVerifyAssertionResponse *)response {
```

Comment:
there're various cases we need to generate a OAuthCredential from verifyAssertion. So we create a method here.